src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,264 @@
+package duke.parser;
+
+import duke.exceptions.InvalidDeadlineException;
+import duke.exceptions.InvalidEventException;
+import duke.exceptions.InvalidKeyException;
+import duke.exceptions.InvalidRequestException;
+import duke.tasks.Event;
+import duke.tasks.Deadline;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.tasks.Todo;
+import duke.storage.Storage;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Interpret the command and do the instruction.
+ */
+public class Parser {
+
+    public static String logo = "      ____        _        \n"
+            + "     |  _ \\ _   _| | _____ \n"
+            + "     | | | | | | | |/ / _ \\\n"
+            + "     | |_| | |_| |   <  __/\n"
+            + "     |____/ \\__,_|_|\\_\\___|\n";
+
+    static TaskList taskList;
+
+    /**
+     * Construct a new paerser.
+     * No parameter. Assign a new TaskList to taskList.
+     */
+    public Parser(){
+        taskList = new TaskList();
+    }
+
+    /**
+     * Print out the greetings.
+     */
+    public static void greet(){
+
+        System.out.println("Hello from\n" + logo);
+
+        System.out.println("What can I do for you?");
+
+    }
+
+    /**
+     * Print out the goodbye words.
+     */
+    public static void exit(){
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Print out the command received.
+     * @param command Take in a command.
+     */
+    public static void echo(String command){
+        System.out.println(command);
+    }
+
+    /**
+     * Process a command sentence.
+     * @param command Take in the command to be processed.
+     * @throws Exception Throws an exception if the command can not be interpreted.
+     */
+    public static void processCommand(String command) throws Exception {
+
+        if(command.equals("list")){
```

Comment:
Can consider reorganising the code so that it is easier to follow. Perhaps consider using switch statements also?